### PR TITLE
Bump `phpspreadsheet`

### PR DESCRIPTION
**Pull request by mjansenDatabay**
If approved, the commits MUST be picked to `trunk` as well.

Unfortunately, the new major version seems to be incompatible with PHP 7.4, so it will not work for `release_8`.

### CODE REVIEW

**Comment by dsstrassner** [comment]
Hi @kergomard, could you look into the T&A related files?

File: Modules/TestQuestionPool/classes/class.ilAssExcelFormatHelper.php [hunk 1]
```diff
@@ -70,12 +69,11 @@ public function setCellByCoordinates($a_coords, $a_value): void
     public function setCell($a_row, $a_col, $a_value, $datatype = null): void
     {
         if (is_string($a_value) && !is_numeric($a_value)) {
-            $this->workbook->getActiveSheet()->setCellValueExplicitByColumnAndRow(
-                $a_col + 1,
-                $a_row,
+            $coordinate = $this->getCoordByColumnAndRow($a_col, $a_row);
```
Comment (mjansenDatabay) [inline Modules/TestQuestionPool/classes/class.ilAssExcelFormatHelper.php:72]: The `+1` on the column index is applied by a call to `columnIndexAdjustment` in `getCoordByColumnAndRow`.

File: Services/Excel/classes/class.ilExcel.php [hunk 2]
```diff
@@ -260,33 +260,29 @@ public function setCell(
         $a_value,
         ?string $a_datatype = null
     ): void {
-        $col = $this->columnIndexAdjustment($a_col);
+        $coordinate = $this->getCoordByColumnAndRow($a_col, $a_row);
```
Comment (mjansenDatabay) [inline Services/Excel/classes/class.ilExcel.php:263]: The `columnIndexAdjustment` is called in `getCoordByColumnAndRow`.

File: Services/Excel/classes/class.ilExcel.php [hunk 3]
```diff
@@ -327,8 +323,8 @@ public function getCell(
         int $a_row,
         int $a_col
     ) {
-        $col = $this->columnIndexAdjustment($a_col);
-        return $this->workbook->getActiveSheet()->getCellByColumnAndRow($col, $a_row)->getValue();
+        $coordinate = $this->getCoordByColumnAndRow($a_col, $a_row);
```
Comment (mjansenDatabay) [inline Services/Excel/classes/class.ilExcel.php:326]: The `columnIndexAdjustment` is called in `getCoordByColumnAndRow`.

File: Services/Excel/classes/class.ilExcel.php [hunk 4]
```diff
@@ -534,8 +530,8 @@ public function addLink(
         int $a_column,
         string $a_path
     ): void {
-        $column = $this->columnIndexAdjustment($a_column);
-        $this->workbook->getActiveSheet()->getCellByColumnAndRow($column, $a_row)->getHyperlink()->setUrl($a_path);
+        $coordinate = $this->getCoordByColumnAndRow($a_column, $a_row);
```
Comment (mjansenDatabay) [inline Services/Excel/classes/class.ilExcel.php:533]: The `columnIndexAdjustment` is called in `getCoordByColumnAndRow`.

**Comment by kergomard** [comment]
Thank you very much @mjansenDatabay !

This looks right and a few tests ran as expected. As `ilExcel` is unmaintained and the other changes are libraries and Test & Assessment. I merge.

Best,
@kergomard

**Pull request merged**